lms/djangoapps/instructor/views/api.py

```diff
@@ -2625,24 +2628,68 @@ def enable_certificate_generation(request, course_id=None):
     return redirect(_instructor_dash_url(course_key, section='certificates'))
 
 
-#---- Gradebook (shown to small courses only) ----
-@cache_control(no_cache=True, no_store=True, must_revalidate=True)
-@require_level('staff')
-def spoc_gradebook(request, course_id):
+def calculate_page_info(offset, total_students):
```

Comment:
Eek, this file is getting insanely long. And there's a lot of code to do w/ gradebook. I wonder if it makes sense - since this code is isolated wrt the code in the rest of the file - to pull it out into its own file within the lms/djangoapps/instructor/views/ directory (maybe `lms/djangoapps/instructor/views/gradebook_api.py`?)